### PR TITLE
fix #290058 translate palette drag painter coords

### PR DESCRIPTION
**Pull request by ericfont**
If the default layout of a palette element resulted in a boundary box with negative x & y coordinates for the top-left corner, the deleted lines of this commit would attempt to ensure that the entire element's pixelmap would be rendered in positive coordinate space only by adjusting accoring to the top-left corner.  Unfortunately, that code don't seem to work, since vertically-cenetered elements like hairpins (which have a negative y coordinate for the top of its boundary box) only display the elements bottom-half when dragged from palette.

This fix uses alternate code which I think more properly does what the deleted lines *intended* to do, but uses Qt's QPainter function for translate instead to offset the painting by the top-left coordinate.

### CODE REVIEW

**Comment by gilbertohasnofb** [comment]
@ericfont This PR seems to fix nearly all objects, but some still have issues. Some of the lines work fine (e.g. first time bracket, second time bracket) but others are still cut. The ones I noted are slurs and hairpins, but there are potentially others.

**Comment by ericfont** [comment]
ahh...yes...it seems my PR has messed up clefs...only draws the top half of clefs now and cuts off the bottom...

**Comment by gilbertohasnofb** [comment]
When I tested, the clefs were fine, I just noticed the issue with the hairpins and slurs.

**Comment by ericfont** [comment]
Fixed that issue with clefs with my recent update of this PR's commit.

Please test with this change.  All palette elements I'm able to drag now are completely rendered when dragged.

**Comment by gilbertohasnofb** [comment]
@ericfont @dmitrio95 I see that this has been merged but the issue hasn't been fixed yet. I just pulled and compiled the current master (after the merge) and I still see the problem I described with the lines. Here are some screenshots. As I wrote, most objects have been fixed, e.g.:

musescore-0
musescore-1

But most lines are still broken. Here are the palette drag icons of the hairpins:

musescore-2
musescore-3

Slurs are completely invisible as drag icons (my mouse is just above the staff in the first bar):

musescore-4

Here are other examples of broken lines:

musescore-5
musescore-6
musescore-7

It seems to me that the only lines whose drag previews are working fine now are the voltas:

musescore-8

**Comment by MarcSabatella** [comment]
Which build specifically (Help / About)?  I just tried and it works fine for me.  Perhaps you didn't wait long enough for the updated build to become available?

**Comment by gilbertohasnofb** [comment]
@MarcSabatella I built it myself after pulling the most recent master into my fork. If it is working on your side then perhaps I've done something wrong then.

**Comment by MarcSabatella** [comment]
Could be, are you experienced at building and running MuseScore from source or was this your first try?  I guess make sure you see the changes in palette.cpp, make sure that file really is getting recompiled, maybe make some other more obvious change to be sure you are really running the version you just built, etc.

**Comment by ericfont** [comment]
@gilbertohasnofb, I just downloaded lastest built nightly for windows (https://ftp.osuosl.org/pub/musescore-nightlies/windows/MuseScoreNightly-2019-07-03-0835-master-dc1e975-x86_64.7z) and my fix seems to be working.  Check 'git log' and make sure my commit is in it, and maybe do a force rebuild...

**Comment by gilbertohasnofb** [comment]
@MarcSabatella my apologies, that was a mistake of mine and everything is fine indeed. The built went fine but I used the wrong `PREFIX` during the installation. Because of that I was actually running an older build from some weeks ago. I have now created a small script to handle this.

TL;DR all fine, my mistake. Apologies for the noise.
